### PR TITLE
Add optional hdfs.confDir config

### PR DESCRIPTION
**Pull request by sfeng88**
When graph.hdfs.confDir is passed in, all properties inside of the directory will be returned as part of the hadoop configuration.

### CODE REVIEW

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java [hunk 1]
```diff
@@ -105,6 +119,37 @@ private URI getHdfsRootDir() throws URISyntaxException {
             Map.Entry entrySet = (Map.Entry) entrySetObject;
             configuration.set("" + entrySet.getKey(), "" + entrySet.getValue());
         }
+
+        String hadoopConfDir = getString(HADOOP_CONF_DIR, null);
```
Comment (joeferner) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java]: can you move all these lines to a method called load hadoop configuration or something.

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java [hunk 2]
```diff
@@ -105,6 +119,37 @@ private URI getHdfsRootDir() throws URISyntaxException {
...
+            LOGGER.info("hadoop conf dir", hadoopConfDir);
+            File dir = new File(hadoopConfDir);
+            if (dir.isDirectory()) {
+                for (String xmlFilename : HADOOP_CONF_FILENAMES) {
+                    File file = new File(dir, xmlFilename);
+                    if (file.isFile()) {
+                        LOGGER.info("adding resource: %s to Hadoop configuration", file);
+                        try {
+                            ByteArrayInputStream in = new ByteArrayInputStream(FileUtils.readFileToByteArray(file));
+                            configuration.addResource(in);
```
Comment (joeferner) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java]: Can this just be a FileInputStream or does it need to be converted to bytes first?

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java [hunk 3]
```diff
@@ -105,6 +119,37 @@ private URI getHdfsRootDir() throws URISyntaxException {
...
+                    }
+                }
+
+                StringBuilder sb = new StringBuilder();
+                SortedSet<String> keys = new TreeSet<>();
+                for (Map.Entry<String, String> entry : configuration) {
+                    keys.add(entry.getKey());
+                }
+
+                LOGGER.debug("Hadoop configuration:%n%s", sb.toString());
```
Comment (joeferner) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java]: What is `%n`?

Comment (dsingley) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraphConfiguration.java]: platform correct line termination (e.g. a newline)

**Comment by joeferner** [comment]
LGTM, can you squash. Then you can merge to master.

**Pull request merged**